cxx-qt-gen/src/parser/qobject.rs

```diff
@@ -44,187 +38,33 @@ pub struct ParsedQObject {
     pub others: Vec<Item>,
 }
 
-#[derive(Default)]
-pub struct ParsedCxxQtData {
-    /// Map of the QObjects defined in the module that will be used for code generation
-    pub qobjects: HashMap<Ident, ParsedQObject>,
-    /// The namespace of the CXX-Qt module
-    pub namespace: String,
-    /// Any `use` statements end up in the CXX-Qt generated module
-    pub uses: Vec<Item>,
-}
-
-impl ParsedCxxQtData {
-    /// Find the QObjects within the module and adds the keys into the qobjects HashMap
-    pub fn find_qobject_keys(&mut self, items: &[Item]) -> Result<()> {
+impl ParsedQObject {
+    /// Extract either invokables or methods from [syn::ImplItem]'s from each Impl block
+    ///
+    /// These will have come from a impl cxx_qt::QObject<T> block
+    pub fn parse_impl_items(&mut self, items: &[ImplItem]) -> Result<()> {
```

Comment:
Think it makes sense to only have ImplItem here. The way it is split is determining which QObject the item is referring to happens in cxxqtdata, then any simple one liners just edit the qobject. Any that have more parsing to do but already know which qobject it is for then have their method in the qobject.rs.  I guess eventually this will be both invokables and properties parsing will be in qobject.rs

Reply:
Makes sense, any complexity once we know the target qobject can be in an impl QObject. So far these are simple though other than invokables (later properties will likely be complex too).